text/0715-construct-upgrade-support.md

```diff
@@ -0,0 +1,1124 @@
+recreated, DynamoDB global tables being recreated). Recreation of resources leads to data loss for
+stateful resources, and can lead to loss of availability for others.
+
+To address this, CDK team develops a construct upgrade framework designed to simplify adoption while
+minimizing risks. This approach combines guided documentation, automated validations, and deployment
+safeguards to ensure a controlled transition.
+
+## Proposed Solution
+
+In a high-level overview, for each supported migration, the construct upgrade framework provides:
+
+- A comprehensive, step-by-step migration guide enabling users to transition from legacy constructs/modules
+to their modern equivalents. Migration guide follows a standardized structure that includes dependency changes,
+behavioral analysis, risk assessments, and rollback instructions. This ensures consistency, clarity, and safety
+across construct migration. See [Appendix A](#appendix-a-migration-guide) for a detailed breakdown of the proposed
+migration guide format.
+
+- A new CDK CLI command `cdk construct-upgrade --unstable=construct-upgrade` to validate code changes and
+execute deployments safely. For more details on the CLI usage, please visit [CLI Settings](#settings)
+
+```sh
+cdk construct-upgrade --unstable=construct-upgrade [stacks] \
+  --target [id] \
+  [OPTIONS]
+```
+
+With the initial release of the CDK construct upgrade tool, developers would use the migration guide to
+update their CDK code, replacing legacy constructs (e.g., Table) with modern equivalents (e.g., TableV2).
+
+After updating the code to use the new construct, users would use the CDK CLI command `cdk construct-upgrade --unstable=construct-upgrade`
+to validates the existing CDK stack and environment, ensuring they are ready for migration. The tool flags any
+blockers, such as CloudFormation drift. Additionally, it analyzes the CloudFormation change set and validates
+any library-specific configurations. See [Construct Upgrade Validations](#construct-upgrade-validations) for a detailed breakdown of the validations.
+
+In the initial release of the CDK construct upgrade tool, VPC-related constructs upgrade from `aws-cdk-lib/aws-ec2`
+to `@aws-cdk/aws-ec2-alpha` and `Table` to `TableV2` in `aws-cdk-lib/aws-dynamodb` come with enhanced
+construct-specific validations and comprehensive migration guides provided by the CDK team to ensure
+safe deployment. These two construct upgrades are so-called `fully supported` construct upgrade.
+
+We aim to gradually increase the number of `fully supported` constructs. The tool is also designed to be open
+and pluggable, allowing users to support other construct migrations by incorporating their own custom
+validations. This flexibility enables users to extend the tool to cover additional constructs that may not
+yet be officially supported with custom validations, see [Custom Validations](#custom-validations).
+
+## Walkthrough of Construct Upgrade Example
```

Comment:
The example is great. I think it would also be very helpful to add the VPC example as well, since it follows the other flow (refactor).